### DISCUSSION TITLE
GUI for RTSP Simple Server

### DISCUSSION THREAD

**Discussion by Tabarane** [bluenviron/mediamtx#1440] [Questions]
Hi everyone, 

i have the idea of using the API part to create a GUI for users in order to manage , create , edit , add , remove all available settings for the server , for example creating a proxy url from different protocol and get the extract protocol, configure ports, access username password all via GUI, this will be an amazing options for managing large deployment and configuration for large scale where they always change the RTMP url source but the need is always RTSP or WEBRTC etc

im willing to sponsor this project if we can have a solid support for GUI that can do it, something similar to Flussonic Media Server

difference is we dont need the IPTV and all other fancy stuff there but a GUI that can manage RTSP SIMPLE SERVER will take the project to the next step for sure

**Comment by aler9** [comment]
Hello, of course a web UI would be a great improvement, personally i never developed the thing for two reasons:

- i always use the server in backends and control it through the APIs.
- I have time limits and human limits. I think it's better to channel the available resources into delivering a small set of features that is reliable and stable instead of trying to develop everything. Bug fixing will always have precedence on adding new features.

Nonetheless, contributions from the community will always be welcome and reviewed with high priority.

**Comment by Tabarane** [comment]
@aler9 i agree no one should do everything alone, im so much invested in the project and i search for someone who can do this even if will take me to sponsor part of it to make it happen, specially i know some technical part but im not a developer, i can do scripts only :)

maybe we can fix the topic in our discussion and see who else is interested and someone might show up and offer help? after all its all for the opensource and alot of people could use it, i can write down some tasks features for the UI what it should have maybe someone will take it from there or you can add it as enhancement request

**Reply by rob-fi** [reply]
I landed on this discussion topic while looking for some sort of existing UI. I've come to the conclusion that perhaps using RESTool (https://github.com/dsternlicht/RESTool) to create a bare minimum type of API UI might be a first step. I intend to give it a go at some point when I have the patience, and if it's successful perhaps I could share the configuration file for others to use.

I have also in the past pasted large chunks of the OBS API documentation into a certain well known language model, and asked it to generate some JS to poll the API to display statistics. I then took time to work on the actual HTML/CSS UI aspect myself. Could be a fast approach if you're also not a developer and merely a script hacker like myself. I believe there's certain tools built into this code hosting platform that do similar things but I haven't looked into them.

**Reply by TimJay** [reply]
Searching for the same, I came across this recently started project:
https://github.com/bcanfield/mediamtx-connect

(not affiliated, not yet tried)

[a comment by an automated account is not shown]
